src/action/ActionManager.js

```diff
@@ -0,0 +1,301 @@
+        continue; // Action is not active, so no new event
+      }
+      this._activeActionInfos.delete(actionPath);
+      this._notifyListeners(actionPath, false, actionParameters, inputSource);
+    }
+  }
+
+  loadDefaults() {
+    // load all of the input sources and filters needed by the default action maps
+    this.addInputSource("gamepad", new GamepadInputSource());
+    this.addInputSource("keyboard", new KeyboardInputSource());
+    this.addInputSource("mouse", new MouseInputSource());
+
+    this.addFilter("reverse-axis", new ReverseAxisFilter());
+    this.addFilter("reverse-active", new ReverseActiveFilter());
+
+    // load the default action maps
+    for (let actionMapInfo of defaultActionMaps) {
+      this.addActionMap(actionMapInfo.name, new ActionMap([...this.filters], actionMapInfo.url));
+    }
+
+    // activate the initial action map
+    this.switchToActionMaps(defaultActionMaps[0].name);
+
+    // TODO load input path aliases
+  }
+
+  /**
+   * @type {Iterator} over [{string}, {@link Filter}] items
+   */
+  get filters() {
+    return this._filters.entries();
+  }
+
+  /**
+   * @param {string} semanticName like 'reverseAxis'
+   * @param {Filter} filter
+   */
+  addFilter(semanticName, filter) {
+    this._filters.set(`/filter/${semanticName}`, filter);
+  }
+
+  /** @type {Iterator} over [{string}, {@link InputSource}] items */
+  get inputSources() {
+    return this._inputSources.entries();
+  }
+
+  /**
+   * @param {string} inputPathName a single element in the semantic path for the input source, like 'keyboard' for a KeyboardInputSource
+   * @param {InputSource} inputSource
+   */
+  addInputSource(inputPathName, inputSource) {
+    this._inputSources.set(`/input/${inputPathName}`, inputSource);
+  }
+
+  /**
+   * Add a path alias for an semantic input path, usually creating a more abstract alias for a specific input.
+   * For example, one might alias `/input/vive/controller/left/button/0` to `/input/hand/left/finger/0`
+   * @param {string} inputPath a full semantic path for the input source
+   * @param {string} aliasPath the alias for the inputPath
+   */
+  aliasInputPath(inputPath, aliasPath) {
+    throw new Error("Not implemented");
+  }
+
+  /** @type {Iterator} over [mapName {string}, {@link ActionMap}] items */
+  get actionMaps() {
+    return this._actionMaps.entries();
+  }
+
+  /** @type {Iterator} over [mapName {string}, {@link ActionMap}] items */
+  get activeActionMaps() {
+    return this._activeActionMaps.entries();
+  }
+
+  /**
+   * @param {string} name
+   * @param {ActionMap} actionMap
+   */
+  addActionMap(name, actionMap) {
+    this._actionMaps.set(name, actionMap);
+  }
+
+  /**
+   * @param {string} name
+   * @return {bool} true if it existed and is removed
+   */
+  removeActionMap(name) {
+    return this._actionMaps.delete(name);
+  }
+
+  /**
+   * Activate named {@link ActionMap}s.
+   * @param {string} names
+   */
+  switchToActionMaps(...names) {
```

Comment:
I think it could be nice to add `activateActionMap` and `deactivateActionMap` that won't clear the current active action maps